### DISCUSSION TITLE
LDAP authentication

### DISCUSSION THREAD

**Discussion by manu0401** [rustdesk/rustdesk#4832] [Ideas]
We consider using RustDesk to control machines in a teaching computer lab, but the unique static password authentication is of some concern. At some point it will leak and be used for student pranks.

It would be nice if RustDesk could have LDAP authentication. I am willing to contribute the code for that if the project is considered reasonable. 

The remote machine would need to authenticate a login+password against LDAP, that seems the easy part.

On the local machine, we would need an UI change so that we can choose authentication method;: static password or login+password.

I have not looked at the protocol between local and remote, and how we could easily cram a login along with the password and maintain  backward compatibility. If no packet versioning is available,  I imagine we can at least start a password field by a magic value to announce it contains a login and a password.

Thoughts? Opinions?

**Comment by kintaro1981** [comment]
I'm seeing this in the Rustdesk subscription Business Plan... is this included in the opensource too?

**Reply by dinger1986** [reply]
No, Pro only

**Comment by manu0401** [comment]
On Mon, Aug 21, 2023 at 10:03:46AM -0700, Kintaro Oe wrote:
> I'm seeing this in the Rustdesk subscription Business Plan... is this included in the opensource too?

I guess that explains the low interest for such a free contribution :-)

-- 
Emmanuel Dreyfus
***@***.***

**Comment by meaz** [comment]
@manu0401 I would be very interested in! We were thinking about offering Rustdesk to our users, but we want the service do be behind ldap so that we don't get too much abuse of the service...
